### DISCUSSION TITLE
[Bug]: Pasting images from the clipboard

### DISCUSSION THREAD

**Discussion by fuegovic** [danny-avila/LibreChat#1851] [Troubleshooting] [answered]
### What happened?

When pasting an image from the clipboard into the chat, the first time it works fine
Doing it a second time within the same convo will resend the first image instead of using the new one

No error about it in the logs, when using assistants I see that the image names are "image.png" both time

side note, the aspect ratio of the pasted screenshot is also slightly wrong

### Steps to Reproduce

- grab a screenshot
- paste it in the chat (from the clipboard)
- wait for the response
- grab a second screenshot
- paste it in the chat (from the clipboard again)
- the first screenshot is being resent instead of the new one

### What browsers are you seeing the problem on?

Microsoft Edge

### Screenshots

![image](https://github.com/danny-avila/LibreChat/assets/32828263/3a1cdd02-523e-48c2-8f1d-87007ef597d1)

These were supposed to be 2 different images

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by danny-avila** [comment]
I can't recreate this on firefox or edge: 
![image](https://github.com/danny-avila/LibreChat/assets/110412045/7819d304-9263-4c6d-b0b5-a1b356712aeb)

**Reply by fuegovic** [reply]
You're right I just tested on the HF demo and it works fine, it only happens on my portainer install 🫤

**Reply by danny-avila** [reply]
weird, I just tested both with firebase strategy and local, still can't reproduce. it'd be interesting to see the server-side logs for when you paste the image (before and after)

**Reply by fuegovic** [reply, marked as answer]
I had firebase in the `.env`, but not in the `yaml`, adding back `fileStrategy: "firebase"` in the yaml config fixed it!

**Comment by fuegovic** [comment]
@danny-avila 
Someone else reported that issue and I was able to reproduce on my end this time.

Without firebase enabled, pasting images from the clipboard seems to overwrite the previously pasted images from the clipboard (which are all named image.png)

see the thread on discord: https://discord.com/channels/1086345563026489514/1215096101468774480

**Reply by fuegovic** [reply]
I was only able to reproduce with local strategy, ~~It works with firebase on my end~~ edit: not quite, see my comment below

**Reply by nidasfly** [reply]
Librechat Server running on Docker, with Mongo DB running on Docker.
Reproduced in Windows Chrome, iOS safari. Only happens when pasting image into the chatbox. All images turned into image.png
![image](https://github.com/danny-avila/LibreChat/assets/56987501/05a02143-5c4e-4b6c-82e9-04cffbf1db2f)
Maybe hash compare or rename strategy is needed.

**Reply by fuegovic** [reply]
even with firebase, once I refreshed the page all displayed images were the same (the last one I uploaded)

**Reply by fuegovic** [reply]
And even with firebase, pasting 2 images from the clipboard sent the same image twice
![image](https://github.com/danny-avila/LibreChat/assets/32828263/61523be9-1ebf-45b3-9202-e2fe6660b46a)

![image](https://github.com/danny-avila/LibreChat/assets/32828263/9d52899a-ff09-4c66-9354-9987de1620a7)

**Reply by danny-avila** [reply]
>  Maybe hash compare or rename strategy is needed.

The images can have the same filenames, they are unique based on ID, not name.

> And even with firebase, pasting 2 images from the clipboard sent the same image twice

I can't reproduce this on any strategy, how are you pasting 2 images from the clipboard?

**Reply by nidasfly** [reply]
So what I did is paste 1 image into the input box, then take another screenshot, paste it into the input box again. This is what shows in the Database.
![image](https://github.com/danny-avila/LibreChat/assets/56987501/c02913bf-9f99-4433-902c-947e702b00b1)
Above is the 3 different images pasted in to chatbox. In fact the 3 images can be stored in the database with different ID. However, when you click into it and check the detail, you can see all their path is the same, which is only 1 image is saved in the server.
![image](https://github.com/danny-avila/LibreChat/assets/56987501/fb122386-a2f8-4bfa-a3ab-661dfec0c418)
![image](https://github.com/danny-avila/LibreChat/assets/56987501/61c9ee26-783d-4c6d-ba6a-709e4adc75d0)

**Reply by fuegovic** [reply]
@danny-avila 
> I can't reproduce this on any strategy, how are you pasting 2 images from the clipboard?

I do this a lot on discord. I take a screenshot with <kbd>Windows</kbd> + <kbd>Shift</kbd> + <kbd>S</kbd> then paste it in the chat input box in LibreChat with <kbd>Ctrl</kbd> + <kbd>V</kbd>

I was able to reproduce the behavior on 2 of my PC and on the HF demo

**Reply by danny-avila** [reply]
thanks guys, handling this here: https://github.com/danny-avila/LibreChat/pull/2015

**Reply by fuegovic** [reply]
Thanks!
